File: java/src/com/kalebklein/Barista.java
/**
 * Barista Java Example
 *
 * Coded in Atom: https://atom.io/
 * License: NONE
 * Repo: https://github.com/pazuzu156/barista
 */

package com.kalebklein;

/**
 * Barista object
 *
 * @author Kaleb Klein
 */
public class Barista
{
	private String str1, str2, str3;

	/**
	 * Class constructor
	 *
	 * @return void
	 */
	public Barista()
	{
		this.str1 = "ers";
		this.str2 = new StringBuilder("rap").reverse().toString();
		this.str3 = "amet";
	}

	/**
	 * Processes request
	 *
	 * @param String preference
	 * @return String
	 */
	public String request(String preference)
	{
		StringBuilder secret = new StringBuilder();
		secret.append(str2).append(str3).append(str1);

		StringBuilder res = new StringBuilder();
		res.append(preference).append(". Secret code: ").append(secret);
		return res.toString();
	}
}

File: php/classes/Barista.php
<?php

/**
 * Barista PHP Example
 *
 * Coded in Atom: https://atom.io/
 * License: NONE
 * Repo: https://github.com/pazuzu156/barista
 */

namespace Classes;

/**
 * Barista Object
 *
 * @package Classes
 */
class Barista
{
	/**
	 * First string
	 *
	 * @var string
	 */
	private $_str1;

	/**
	 * Second string
	 *
	 * @var string
	 */
	private $_str2;

	/**
	 * Third string
	 *
	 * @var string
	 */
	private $_str3;

	/**
	 * Class constructor
	 *
	 * @return void
	 */
	public function __construct()
	{
		$this->_str1 = 'ers';
		$this->_str2 = strrev('rap');
		$this->_str3 = 'amet';
	}

	/**
	 * Processes the request
	 *
	 * @param string $preference
	 * @return string
	 */
	public function request($preference)
	{
		return $preference . '. Secret word: '
		 . $this->_str2 . $this->_str3 . $this->_str1;
	}
}
